Sources/SQLKit/Builders/Implementations/SQLInsertBuilder.swift

```diff
@@ -0,0 +1,384 @@
+    ///   - userInfo: See ``SQLQueryEncoder/userInfo``.
+    @inlinable
+    @discardableResult
+    public func model(
+        _ model: some Encodable,
+        prefix: String? = nil,
+        keyEncodingStrategy: SQLQueryEncoder.KeyEncodingStrategy = .useDefaultKeys,
+        nilEncodingStrategy: SQLQueryEncoder.NilEncodingStrategy = .default,
+        userInfo: [CodingUserInfoKey: any Sendable] = [:]
+    ) throws -> Self {
+        try self.models(
+            [model],
+            prefix: prefix,
+            keyEncodingStrategy: keyEncodingStrategy,
+            nilEncodingStrategy: nilEncodingStrategy,
+            userInfo: userInfo
+        )
+    }
+
+    /// Use an `Encodable` value to generate a row to insert and add that row to the query.
+    ///
+    /// Example usage:
+    ///
+    /// ```swift
+    /// let earth = Planet(id: nil, name: "Earth", isInhabited: true)
+    /// let encoder = SQLQueryEncoder(nilEncodingStrategy: .asNil)
+    ///
+    /// try await sqlDatabase.insert(into: "planets")
+    ///     .model(earth, with: encoder)
+    ///     .run()
+    ///
+    /// // Effectively the same as:
+    /// try await sqlDatabase.insert(into: "planets")
+    ///     .columns("id", "name", "isInhabited")
+    ///     .values(SQLLiteral.null, SQLBind(earth.name), SQLBind(earth.isInhabited))
+    ///     .run()
+    /// ```
+    ///
+    /// > Note: The term "model" does _not_ refer to Fluent's `Model` type. Fluent models are not compatible with
+    /// > this method or any of its variants.
+    ///
+    /// - Parameters:
+    ///   - model: A value to insert. This can be any encodable type which represents an aggregate value.
+    ///   - encoder: A preconfigured ``SQLQueryEncoder`` to use for encoding.
+    @inlinable
+    @discardableResult
+    public func model(
+        _ model: some Encodable,
+        with encoder: SQLQueryEncoder
+    ) throws -> Self {
+        try self.models([model], with: encoder)
+    }
+
+    /// Use an array of `Encodable` values to generate rows to insert and add those rows to the query.
+    ///
+    /// Example usage:
+    ///
+    /// ```swift
+    /// let earth = Planet(id: nil, name: "Earth", isInhabited: true)
+    /// let mars = Planet(id: nil, name: "Mars", isInhabited: false)
+    ///
+    /// try await sqlDatabase.insert(into: "planets")
+    ///     .models([earth, mars], keyEncodingStrategy: .convertToSnakeCase)
+    ///     .run()
+    ///
+    /// // Effectively the same as:
+    /// try await sqlDatabase.insert(into: "planets")
+    ///     .columns("id", "name", "is_inhabited")
+    ///     .values(SQLBind(earth.id), SQLBind(earth.name), SQLBind(earth.isInhabited))
+    ///     .values(SQLBind(mars.id), SQLBind(mars.name), SQLBind(mars.isInhabited))
+    ///     .run()
+    /// ```
+    ///
+    /// > Note: The term "model" does _not_ refer to Fluent's `Model` type. Fluent models are not compatible with
```

Comment:
IMO this isn't needed, this package can be used without Fluent too so people who might do that could be confused